WordPress/src/main/java/org/wordpress/android/viewmodel/giphy/provider/TenorProvider.kt

```diff
@@ -50,45 +57,78 @@ internal class TenorProvider @JvmOverloads constructor(
     }
 
     /**
-     * This method act as a simplification to call a search within the Tenor API and the callback
-     * creation
-     * [MediaFilter] must be BASIC or the returned Media will not have a displayable thumbnail
-     * All other provided parameters are set following the Tenor API Documentation
+     * The [onSuccess] will be called if the response is not null
      *
      * The [onFailure] will be called assuming that no valid GIF was found
      * or that a direct failure was found by Tenor API
      *
      * Method is inlined for better high-order functions performance
      */
-    private inline fun IApiClient.simpleSearch(
+    private inline fun IApiClient.enqueueSearchRequest(
         query: String,
         position: String,
         loadSize: Int?,
         crossinline onSuccess: (GifsResponse) -> Unit,
         crossinline onFailure: (Throwable?) -> Unit
-    ) {
-        search(
-                ApiClient.getServiceIds(context),
-                query,
-                loadSize.fittedToMaximumAllowed,
-                position,
-                MediaFilter.BASIC,
-                AspectRatioRange.ALL
-
-        ).enqueue(object : WeakRefCallback<Context, GifsResponse>(context) {
+    ) = buildSearchCall(query, loadSize, position).apply {
+        enqueue(object : WeakRefCallback<Context, GifsResponse>(context) {
             override fun success(context: Context, response: GifsResponse?) {
+                this@apply.cancel()
```

Comment:
For my own surprise, I've detected that the `WeakRefCallback` wasn't canceling the Call, it was reaching the `success` and `failure` with the `isCanceled` returning false. But after the manual timeout implementation removal, this kind of handling lost its purpose since we now rely on the default retrofit Callback implementation. I've fixed this point in f6123f6993b77a68deeec3db38576f2024c4b000. I've also added below a comment given further context about my solution and how I decided to proceed.